### PR TITLE
confluent-platform: disable due to OSI license issue

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
It is using `https://www.confluent.io/confluent-community-license/`

```
Is Confluent Community License open source?
Strictly speaking it is “source-available.” Many people use the phrase “open source” in a loose sense to mean that you can freely download, modify, and redistribute the code, and those things are all true of the code under the Confluent Community License. However, in the strictest sense “open source” means a license approved by the Open Source Initiative (“OSI”) which meets a particular set of criteria. The Confluent Community License is not approved by the OSI and likely would not be as it excludes the use case of creating a SaaS offering of the code. Because of this, we will not refer to the Confluent Community License or any code released under it as open source.
```

**Comment by Rylan12** [comment]
I think this states pretty clearly that the software is not really "open source."

As indicated [in the docs](https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae):

> Our policy is that formulae in the core tap ([homebrew/core](https://github.com/Homebrew/homebrew-core)) must be open-source with an [Debian Free Software Guidelines license](https://wiki.debian.org/DFSGLicenses)

To me, this means `confluent-platform` should be removed.

**Comment by dtrodrigues** [comment]
This sounds like a more restrictive license than MongoDB's SSPL, which also was removed from Homebrew.

**Comment by SMillerDev** [comment]
Should probably disable it to notify people, rather then plain removal.

**Comment by chenrui333** [comment]
Yeah, I would vote for disable rather than plain removal (more smooth user experience transition)

**Comment by chenrui333** [comment]
Proposing to disable the formula due to the OSI license issue.

File: Formula/confluent-platform.rb [hunk 1]
```diff
@@ -12,6 +12,8 @@ class ConfluentPlatform < Formula
 
   bottle :unneeded
 
+  disable! because: :"OSI license issue"
```
Comment (Rylan12) [inline Formula/confluent-platform.rb:15]:
```suggestion
  disable! because: "does not have an OSI license"
```

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**